assets/js/modules/reader-revenue-manager/components/common/PublicationSelect.test.js

```diff
@@ -0,0 +1,150 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import PublicationSelect from './PublicationSelect';
+import { MODULES_READER_REVENUE_MANAGER } from '../../datastore/constants';
+import { publications } from '../../datastore/__fixtures__';
+import {
+	act,
+	createTestRegistry,
+	fireEvent,
+	render,
+	provideUserAuthentication,
+} from '../../../../../../tests/js/test-utils';
+
+/* eslint-disable sitekit/acronym-case */
+describe( 'PublicationSelect', () => {
+	let registry;
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+		const { dispatch } = registry;
+
+		provideUserAuthentication( registry );
+
+		dispatch( MODULES_READER_REVENUE_MANAGER ).receiveGetPublications(
+			publications
+		);
+
+		dispatch( MODULES_READER_REVENUE_MANAGER ).setPublicationID(
+			publications[ 0 ].publicationId
+		);
+	} );
+
+	it( 'should return null if the publication ID is invalid.', async () => {
+		// Set the invalid publication ID.
+		registry
+			.dispatch( MODULES_READER_REVENUE_MANAGER )
+			.setPublicationID( 'ABCD*&^' );
+
+		const { container, waitForRegistry } = render(
+			<PublicationSelect isDisabled={ false } />,
+			{
+				registry,
+			}
+		);
+
+		await waitForRegistry();
+
+		// Component must return `null` for invalid publication ID.
+		expect( container.firstChild ).toBeNull();
+		expect( container ).toBeEmptyDOMElement();
+
+		act( () => {
+			registry
+				.dispatch( MODULES_READER_REVENUE_MANAGER )
+				.setPublicationID( publications[ 0 ].publicationId );
+		} );
+
+		// After we set a valid publication ID, the publication select should be visible.
+		expect(
+			container.querySelector(
+				'.googlesitekit-analytics-4__select-publication'
+			)
+		).toBeInTheDocument();
+
+		expect(
+			container.querySelector( '.mdc-select__selected-text' )
+		).toBeInTheDocument();
+	} );
+
+	it( 'should render an option for each publication.', async () => {
+		const { getAllByRole, waitForRegistry } = render(
+			<PublicationSelect isDisabled={ false } />,
+			{
+				registry,
+			}
+		);
+
+		await waitForRegistry();
+
+		const listItems = getAllByRole( 'menuitem', { hidden: true } );
+		expect( listItems ).toHaveLength( publications.length );
+	} );
+
+	it( 'should disable the public select if the user does not have module access', async () => {
+		const { container, getAllByRole, waitForRegistry } = render(
+			<PublicationSelect
+				isDisabled={ false }
+				hasModuleAccess={ false }
+			/>,
+			{
+				registry,
+			}
+		);
+
+		await waitForRegistry();
+
+		const listItems = getAllByRole( 'menuitem', { hidden: true } );
+		expect( listItems ).toHaveLength( 1 );
+
+		// Verify that the Publication select dropdown is disabled.
+		[
+			'.googlesitekit-analytics-4__select-publication',
```

Comment:
This is an unrelated class name which has been proposed to be removed. Let's remove it here as well. If it is necessary to persist, let's change it to something like `.googlesitekit-rrm__select-publication`.